Frends.PostgreSQL.ExecuteQuery/Frends.PostgreSQL.ExecuteQuery/ExecuteQuery.cs

```diff
@@ -0,0 +1,128 @@
+    /// </summary>
+    /// <param name="input">Query, parameters and connection string.</param>
+    /// <param name="options">Set timeout and isolation level.</param>
+    /// <param name="cancellationToken">Automatically generated and passed by Frends.</param>
+    /// <returns>Result of the query. JToken QueryResult</returns>
+    public static async Task<Result> ExecuteQuery([PropertyTab] Input input, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        using (var conn = new NpgsqlConnection(input.ConnectionString))
+        {
+            await conn.OpenAsync(cancellationToken);
+            var transaction = conn.BeginTransaction(GetIsolationLevel(options.SqlTransactionIsolationLevel));
+            using (var cmd = new NpgsqlCommand(input.Query, conn))
+            {
+                cmd.CommandTimeout = options.CommandTimeoutSeconds;
+                cmd.Transaction = transaction;
+
+                // Add parameters to command, if any were given.
+                if (input.Parameters != null && input.Parameters.Length > 0)
+                {
+                    foreach (var parameter in input.Parameters)
+                    {
+                        cancellationToken.ThrowIfCancellationRequested();
+
+                        // Convert parameter.Value to DBNull.Value if it is set to null.
+                        if (parameter.Value == null)
+                        {
+                            cmd.Parameters.AddWithValue(parameter.Name, DBNull.Value);
+                        }
+                        else
+                        {
+                            cmd.Parameters.AddWithValue(parameter.Name, parameter.Value);
+                        }
+                    }
+                }
+
+                // Execute command.
+
+                if (input.Query.ToLower().Contains("select"))
+                {
+                    var reader = await cmd.ExecuteReaderAsync(cancellationToken);
+                    var result = reader.ToJson();
+                    conn.Close();
+                    return new Result(result);
+                }
+                else
+                {
+                    var rows = await cmd.ExecuteNonQueryAsync(cancellationToken);
+                    transaction.Commit();
+                    conn.Close();
+                    return new Result(JToken.FromObject(new { AffectedRows = rows }));
+                }
+            }
+        }
+    }
+
+    #region HelperMethods
+
+    // Extension method for NpgsqlDataReader to read the data and return it as JToken.
+    private static JToken ToJson(this NpgsqlDataReader reader)
+    {
+        // Create JSON result.
+        using (var writer = new JTokenWriter())
+        {
+            writer.Formatting = Newtonsoft.Json.Formatting.Indented;
+            writer.Culture = CultureInfo.InvariantCulture;
+
+            // Start array.
+            writer.WriteStartArray();
+
+            while (reader.Read())

```

Comment:
Should cancellationToken be given to this to ensure that the loop doesn't end up as infinityloop?